### PR TITLE
TAR-72 Error handling when Auth service returns 401

### PR DESCRIPTION
**Pull request by tamaslang**
Handle auth 401 with redirecting to the login page. Previously it was resolved to an Internal Server Error

### CODE REVIEW

File: app/auth/RouterAuthenticationProvider.scala [hunk 1]
```diff
@@ -35,9 +35,12 @@ trait RouterAuthenticationProvider extends AuthenticationProvider {
   override def redirectToLogin(implicit request: Request[_]): Future[FailureResult] = Future.successful(Redirect(login))
 
   override def handleNotAuthenticated(implicit request: Request[_]): PartialFunction[UserCredentials, Future[Either[AuthContext, RouterAuthenticationProvider.FailureResult]]] = {
-    case UserCredentials(None, token@_) =>
-      Logger.info(s"No userId found - redirecting to login. user: None token : $token")
-      redirectToLogin.map(Right(_))
+   case UserCredentials(None, token@_) =>
```
Comment (anshulbajpai) [inline app/auth/RouterAuthenticationProvider.scala]: The pattern matching should have specific cases defined first.

File: app/auth/RouterAuthenticationProvider.scala [hunk 2]
```diff
@@ -35,9 +35,12 @@ trait RouterAuthenticationProvider extends AuthenticationProvider {
...
+     Logger.info(s"No userId found - redirecting to login. user: None token: $token")
```
Comment (anshulbajpai) [inline app/auth/RouterAuthenticationProvider.scala]: What is different happening in these two cases?

**Comment by anshulbajpai** [comment]
No unit test? Is it possible to unit test it as well?

**Comment by tamaslang** [comment]
I close this one off, I can not reproduce the problem in the original ticket.

File: app/auth/RouterAuthenticationProvider.scala (hunk 2 above)
Comment (scottkwalker) [inline app/auth/RouterAuthenticationProvider.scala]: the case below seems to be redundant

**Pull request closed without merging**